app/views/RegisterView.js

```diff
@@ -35,15 +39,30 @@ export default class RegisterView extends React.Component {
 	static propTypes = {
 		navigation: PropTypes.object,
 		loginRequest: PropTypes.func,
-		Site_Name: PropTypes.string
+		Site_Name: PropTypes.string,
+		Accounts_CustomFields: PropTypes.string
 	}
 
-	state = {
-		name: '',
-		email: '',
-		password: '',
-		username: '',
-		saving: false
+	constructor(props) {
+		super(props);
+		const customFields = {};
+		this.parsedCustomFields = {};
+		if (props.Accounts_CustomFields) {
+			this.parsedCustomFields = JSON.parse(props.Accounts_CustomFields);
```

Comment:
`Accounts_CustomFields` isn't correctly configured, the parse may fail and it'll crash the app.
Can you add a `try/catch`?

Reply:
done